javaslang/src/main/java/javaslang/control/Option.java

```diff
@@ -54,14 +60,14 @@
      */
     static <T> Option<Seq<T>> sequence(Iterable<? extends Option<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        List<T> list = List.empty();
+        Vector<T> vector = Vector.empty();
         for (Option<? extends T> value : values) {
             if (value.isEmpty()) {
                 return Option.none();
             }
-            list = list.prepend(value.get());
+            vector = vector.append(value.get());
         }
-        return Option.some(list.reverse());
+        return Option.some(vector);
     }
```

Comment:
@ashrko619 we can't do that right now. I targeted 3.0.0 for such compatibility breaking changes (see also #1493)